packages/netlify-cms-backend-github/src/implementation.js

```diff
@@ -278,13 +311,55 @@ export default class GitHub {
     }
   }
 
-  updateUnpublishedEntryStatus(collection, slug, newStatus) {
-    return this.api.updateUnpublishedEntryStatus(collection, slug, newStatus);
+  async migrateUnpublishedEntries(entries) {
+    const errors = [];
+    const entriesMigration = entries
+      .map(entry => {
+        return this.api.updateStatusLabel(
+          entry.metaData.pr.number,
+          entry.metaData.status,
+          entry.metaData.status,
+        );
+      })
+      .map(p => p.catch(err => errors.push(err)));
+
+    await Promise.all(entriesMigration);
+
+    if (errors.length > 0) {
+      throw new Error('Error migrating entries');
+    }
+
+    return this.api.deleteMetadata();
+  }
+
+  getStatusLabels(status) {
+    return Map({
+      [status.get('DRAFT')]: Map({ name: 'netlify-cms/draft', color: 'fad8c7' }),
+      [status.get('PENDING_REVIEW')]: Map({ name: 'netlify-cms/review', color: 'fef2c0' }),
+      [status.get('PENDING_PUBLISH')]: Map({ name: 'netlify-cms/publish', color: 'c2e0c6' }),
+    });
+  }
+
+  inferCollection(path) {
+    const collections = this.config.get('collections').toJS();
+    const collection = collections.find(collection => {
+      return (
+        (collection.files && collection.files.some(f => f.file === path)) ||
+        path.indexOf(collection.folder) >= 0
```

Comment:
This will match any file whose path *contains* the folder path rather than just those that *begin* with the folder path - this could easily be a problem, eg. if someone has "posts" as their folder, but then an entry isn't in that collection has "posts" in it's filename, we could infer the wrong collection.

Also just realizing that nested folder collections could be trouble, eg. "posts" and "posts/podcasts" - everything in "post/podcasts" would be be infered as "posts" since they all begin with that string. We could ensure against this by sorting collections folder type first and longest first.